### PR TITLE
update Miri

### LINKED ISSUE #97486: `miri` no longer builds after rust-lang/rust#97158
Hello, this is your friendly neighborhood mergebot.
After merging PR rust-lang/rust#97158, I observed that the tool miri has failing tests.
A follow-up PR to the repository https://github.com/rust-lang/miri is needed to fix the fallout.

cc @JakobDegen, do you think you would have time to do the follow-up work?
If so, that would be great!

### PR DESCRIPTION
**Pull request by RalfJung**
First update with the new ui test suite, let's hope this all works. :)
r? @oli-obk 

Fixes https://github.com/rust-lang/rust/issues/97486

### CODE REVIEW

**Comment by RalfJung** [comment]
Maybe they are incorrectly being optimized?

**Comment by oli-obk** [comment]
> Maybe they are incorrectly being optimized?

just wondered that, too, and found `"-Zmir-opt-level=4"` in the logs...

**Comment by oli-obk** [comment]
Hmm, we only ever ran it with mir opt level 4 apparently, not with zero like our test suite does? https://github.com/rust-lang/rust/blob/855fc022fe879f4e3493a024f9c6b981d6317612/src/bootstrap/test.rs#L576

**Comment by RalfJung** [comment]
> Hmm, we only ever ran it with mir opt level 4 apparently, not with zero like our test suite does? 

We run it the regular way here:
https://github.com/rust-lang/rust/blob/855fc022fe879f4e3493a024f9c6b981d6317612/src/bootstrap/test.rs#L571

File: src/bootstrap/test.rs [hunk 1]
```diff
@@ -572,8 +572,6 @@ impl Step for Miri {
                 return;
             }
 
-            // # Run `cargo test` with `-Zmir-opt-level=4`.
-            cargo.env("MIRIFLAGS", "-O -Zmir-opt-level=4");
             if !try_run(builder, &mut cargo) {
                 return;
             }
```
Comment (RalfJung) [inline src/bootstrap/test.rs:577]: Now we are just running it twice. :)

**Pull request merged**